src/agent/manager.cpp

```diff
@@ -860,22 +864,41 @@ Anonymous* createOverlayAgentManager(const Parameters& parameters)
 
   if (agentConfig.isNone()) {
     LOG(ERROR) << "Missing `agent_config`";
+
     return nullptr;
   }
 
+  Option<string> master = os::getenv(MESOS_MASTER);
+  if (master.isNone() && agentConfig->has_master()) {
+    master = agentConfig->master();
```

Comment:
That was the idea when we discussed this last to last week. However, now that you bring it up it would better to have the module config trump the EnvVar, since in DC/OS we can always introduce the parameter if we ever want the ZK URL of the module and the agent to diverge. Will invert this.